### DISCUSSION TITLE
Should plugins be in scope of the ghc-proposals process?

### DISCUSSION THREAD

**Discussion by adamgundry** [ghc-proposals/ghc-proposals#548] [General]
There seems to be a tacit consensus that GHC plugins are not in scope of the ghc-proposals process. See for example the discussion on #396, and more recently https://gitlab.haskell.org/ghc/ghc/-/issues/22401. It's not clear to me why this is the case though. While the GHC API is not very well bounded at the moment and is rather subject to change with the implementation, stability is particularly valuable for the plugin interfaces because user code is particularly likely to rely on them.

Clearly plugins might be affected by a wide range of changes to the GHC implementation, and we wouldn't necessarily want to make all of them subject to this process. However, for significant changes such as the introduction of entirely new plugin interfaces, I think the process of writing a proposal and having it reviewed can lead to better design. It also helps establish consensus that there is sufficient motivation to justify new features.

Should we change this policy and bring plugins into the process?

**Comment by nomeata** [comment]
My motivation when saying that plugin API changes are out of scope is mostly that I didn't want to slow down innovation on that front. I see the committee process as a hurdle we introduce to somewhat filter the changes and features we unleash on Haskell users. I assumed plugin authors don't need the same level of protection (smaller number, higher level of expertise and - for better or for worse - used to breaking changes), and try to avoid imposing bureaucracy whenever possible.

But if it is the case that plugin authors need to deserve a similar level of protection, or alternatively if the committee process actually improves quality (in addition to filtering), then I see no reason not to handle plugin proposals here.

**Comment by goldfirere** [comment]
My worry is that this group isn't necessarily as plugged in (ha ha) to plugins, and thus we might not be the best decision-makers. What I would love is a group of people actively using plugins to help shape their future. I suppose, lacking that, we could take it on here, but I think that's less-than-ideal.
